File: [Manya]/Leetcode/Sliding-Window/1343.java
import java.util.Scanner;

class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int n = arr.length;
        int count = 0;
        int target = k * threshold;
        int sum = 0;

        for (int i = 0; i < k; i++) {
            sum += arr[i];
        }

        if (sum >= target) {
            count++;
        }

        for (int i = k; i < n; i++) {
            sum += arr[i] - arr[i - k];
            if (sum >= target) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the number of elements in the array: ");
        int n = scanner.nextInt();
        int[] arr = new int[n];

        System.out.println("Enter the elements of the array:");
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }

        System.out.print("Enter the value of k: ");
        int k = scanner.nextInt();

        System.out.print("Enter the threshold: ");
        int threshold = scanner.nextInt();

        Solution solution = new Solution();
        int result = solution.numOfSubarrays(arr, k, threshold);
        System.out.println("Number of subarrays meeting the condition: " + result);

        scanner.close();
    }
}

File: Aditya-Pratap/Sliding Window/1343.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        th = threshold * k 

        res = 0
        # for i in range(k):
        #     avg += arr[i]
        avg=sum(arr[:k])
        if avg >= th:
            res += 1

        for i in range(k, len(arr)):
            avg -= arr[i-k]
            avg += arr[i]
            if avg >= th:
                res += 1

        return res